src/utils/web.ts

```diff
@@ -17,39 +17,39 @@ export interface ConnectionInfo {
     headers?: { [key: string]: string | number };
 }
 
-export async function fetchJson(connection: string | ConnectionInfo, json?: string): Promise<any> {
-    let url: string = null;
-    if (typeof(connection) === 'string') {
-        url = connection;
+export async function fetchJson(connectionInfoOrUrl: string | ConnectionInfo, json?: string): Promise<any> {
+    let connectionInfo: ConnectionInfo = null;
+    if (typeof (connectionInfoOrUrl) === 'string') {
+        connectionInfo.url = connectionInfoOrUrl;
```

Comment:
Sadly, although `typeof null` is `object`, it isn't actually an object and will throw an exception if you try to use like as one -- as far as I am aware, referencing a property of `null` will throw a TypeError (Cannot set property x of null).  I'm actually pretty surprised that Typescript didn't error on compiling this, unless Typescript transpile is converting this weirdly to make the property access safe...

**We should have a test case for this since it is the backwards compatibility branch that all existing consumers of this code will be running when we release it.**

Reply:
Nice catch. And yes, the compiler was silent about it. Fixed, + created test.